scripts/zones/Maze_of_Shakhrami/mobs/Leech_King.lua

```diff
@@ -0,0 +1,23 @@
+-----------------------------------
+-- Area: Maze of Shakhrami
+--  MOB: Leech King (NM)
+-----------------------------------
+
+require("scripts/zones/Maze_of_Shakhrami/MobIDs");
+
+-----------------------------------
+-- onMobDeath
+-----------------------------------
+
+function onMobDeath(mob,killer,ally)
+    local whichNM = math.random(0,19);
+
+    if (whichNM < 10) then
+        SetRespawnTime(Argus, 64800, 108000); -- 18-30 hours
+        DeterMob(Leech_King, true);
+    else
+        SetRespawnTime(Leech_King, 64800, 108000); -- 18-30 hours
+        DeterMob(Argus, true);
```

Comment:
You're right about onZoneInitialize. If you defeat the mob that was chosen on boot, and it gets selected again, there's no need to change what mob is deterred - only need to set the respawn time. If it's different, you deter the mob you just killed, and have the other mob not be deterred.